### ISSUE TITLE
Question: bedtools genomecov -d output from bam files does not include chromosomes with not coverage

### ISSUE THREAD

**Issue by bchapm5** [arq5x/bedtools2#71]
after creating a BAM file by aligning RNA sequence to a small file of coding sequences bedtools genomecov -d did not report sequences with zero coverage over the entire length.

Is this a feature or a bug ?

$samtools idxstats ARC10_.unique.bam > ARC10_.unique.idxstats # get output of idxstats
$cat ARC10_.unique.idxstats | grep -v "^\*" | wc -l # count the number of entries in idxstats
233
$cat ARC10_.unique.idxstats | cut -f3 | grep -v '^0' | wc -l # count the number of entries in idx stats without zero coverage
158
$./bedtools --versionbedtools v2.19.1-41-g656fd84 # show we are using current version of bedtoos
$ ./bedtools genomecov -dz -ibam ARC10_.unique.bam > ARC10_.unique.depth # output all coverage of bedtools
$ cat ARC10_.unique.depth | cut -f1 | uniq | wc -l # count the number of different sequences in bedtools
158

I hope to get small example fasta and SAM file together to better demonstrate I hope the above is understandable

**Comment by marcelm** [comment]
I’m just seeing the same problem in bedtools 2.21.0. To reproduce it:
```
echo -e '@SQ\tSN:ref1\tLN:10' > small.sam
echo -e '@SQ\tSN:ref2\tLN:5' >> small.sam
echo -e 'r1\t0\tref2\t1\t1\t3M\t*\t0\t0\tACG\tFFF' >> small.sam
samtools view -bS small.sam > small.bam
echo -e 'ref1\t10\nref2\t5' > genome.txt
bedtools genomecov -d -ibam small.bam -g genome.txt
```
Result:
```
ref2    1       1
ref2    2       1
ref2    3       1
ref2    4       0
ref2    5       0
```
That is, zero coverage on ref1 is not reported although it is listed in `genome.txt`. This also happens with `-bga`.

I think this is a bug and not a feature: If I use `-d` or `-bga` to get zero coverage regions, then I want *all* those regions. At least that’s what I had expected to get reading the documentation. Would be great to see this fixed!

**Comment by arq5x** [comment]
Yep, this looks like a bug. I will try to fix it asap.

**Comment by marcelm** [comment]
Hi, have you had a chance to look into this?

**Comment by arq5x** [comment]
Sorry for the delayed response.  I have taken a look at the code and have a plan in mind for addressing this, but I just haven't had a free chunk of time to knock this out yet.  I apologize.

**Comment by arq5x** [comment]
I just pushed a fix (commit b588bec) that addresses this. Sorry for the trouble, this should have been fixed long ago but was lost in the mix.  
The only caveat is that uncovered chromosomes are reported after covered chromosomes, so you may need to resort the output if you want a common order.

**Comment by marcelm** [comment]
Saw the Github notification only now, but better late than never: Thanks for fixing this!
